### DISCUSSION TITLE
Balancing current reported wrong when negative value

### DISCUSSION THREAD

**Discussion by marcellinusdeman** [txubelaxu/esphome-jk-bms#36] [General]
There is a conversion problem in the code regarding the balancing current. It reports the wrong value when negative.

[13:28:57][D][sensor:094]: 'rs485-bms0 balancing current': Sending state 64.54401 A with 2 decimals of accuracy

Marcel

**Comment by x230-thinkpad** [comment]
> There is a conversion problem in the code regarding the balancing current. It reports the wrong value when negative.
> 
> [13:28:57][D][sensor:094]: 'rs485-bms0 balancing current': Sending state 64.54401 A with 2 decimals of accuracy
> 
> Marcel



Hi Marcel,

I just notice, I have same issue
it also for positive value, the reading not accurate

![image](https://github.com/user-attachments/assets/07add03f-20a8-4d88-b05b-a34d4c5b5d85)

my bms 2A version, it should be no more than 2A for balancing current, hopefully @txubelaxu have time for fix it. thank you

**Comment by txubelaxu** [comment]
I have started reviwing this issue. Same here:

![image](https://github.com/user-attachments/assets/fc0009ed-07ad-44eb-843c-1727ffdc72a0)


![image](https://github.com/user-attachments/assets/fc57e361-2e0c-4e09-a260-8962061e3808)

![image](https://github.com/user-attachments/assets/27540b46-eeb7-4900-a2d3-35e9bbeb8526)

![image](https://github.com/user-attachments/assets/0af64fa6-b126-45ac-8de2-c60fb8a4d79c)

**Comment by txubelaxu** [comment]
I have just uploaded a change to txubelaxu/esphome-jk-bms with a correction. Could it solve the issue?

**Reply by marcellinusdeman** [reply]
Just updated the code seconds ago. System up and running. Have to wait until tomorrow afternoon when batteries are full to check if the reporting is fine.

Marcel

**Comment by x230-thinkpad** [comment]
> I have just uploaded a change to txubelaxu/esphome-jk-bms with a correction. Could it solve the issue?

![image](https://github.com/user-attachments/assets/04033822-bad9-4ba2-a724-3deed0b14aa0)

it looks normal already. thank you very much

**Comment by marcellinusdeman** [comment]
Yup sam here
